sql/src/main/java/io/crate/planner/DecommissionNodePlan.java

```diff
@@ -52,24 +57,58 @@ public void executeOrFail(DependencyCarrier dependencies,
                               RowConsumer consumer,
                               Row params,
                               SubQueryResults subQueryResults) {
-        final String targetNodeIdOrName = DataTypes.STRING.value(
-            SymbolEvaluator.evaluate(plannerContext.transactionContext(),
-                dependencies.functions(),
-                analyzedDecommissionNodeStatement.nodeIdOrName(),
-                params,
-                subQueryResults)
-        );
+        var boundedDecommissionNode = createStatement(
+            analyzedDecommissionNode,
+            plannerContext.transactionContext(),
+            plannerContext.functions(),
+            params,
+            subQueryResults);
 
-        final String targetNodeId = NodeSelection.resolveNodeId(
-            dependencies.clusterService().state().nodes(), targetNodeIdOrName);
+        String targetNodeId = NodeSelection.resolveNodeId(
+            dependencies.clusterService().state().nodes(),
+            boundedDecommissionNode.nodeIdOrName()
+        );
 
-        OneRowActionListener<AcknowledgedResponse> listener =
-            new OneRowActionListener<>(consumer, r -> r.isAcknowledged() ? new Row1(1L) : new Row1(0L));
+        dependencies
+            .transportActionProvider()
+            .transportDecommissionNodeAction()
+            .execute(
+                targetNodeId,
+                DecommissionNodeRequest.INSTANCE,
+                new OneRowActionListener<>(
+                    consumer,
+                    r -> r.isAcknowledged() ? new Row1(1L) : new Row1(0L))
+            );
+    }
 
-        dependencies.transportActionProvider().transportDecommissionNodeAction().execute(
-            targetNodeId,
-            DecommissionNodeRequest.INSTANCE,
-            listener
+    @VisibleForTesting
+    public static BoundedDecommissionNode createStatement(AnalyzedDecommissionNode decommissionNode,
+                                                          CoordinatorTxnCtx txnCtx,
+                                                          Functions functions,
+                                                          Row parameters,
+                                                          SubQueryResults subQueryResults) {
+        Function<? super Symbol, Object> eval = x -> SymbolEvaluator.evaluate(
```

Comment:
Could avoid the `Function` lambda creation and evaluate `nodeIdOrName` directly as there are no other symbols